cpp/tests/large_strings/replace_tests.cpp

```diff
@@ -65,3 +66,22 @@ TEST_F(StringsManyTest, Replace)
     CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(c, expected);
   }
 }
+
+TEST_F(ReplaceTest, ReplaceWide)
+{
+  auto const expected   = this->long_column();
+  auto const view       = cudf::column_view(expected);
+  auto const multiplier = 10;
+  auto const separator  = cudf::string_scalar("|");
+  auto const input      = cudf::strings::concatenate(
+    cudf::table_view(std::vector<cudf::column_view>(multiplier, view)), separator);
+
+  auto const input_view = cudf::strings_column_view(input->view());
+  auto const target     = cudf::string_scalar("3");  // fake the actual replace;
+  auto const repl       = cudf::string_scalar("3");  // logic still builds the output
```

Comment:
are we certain this will always be the case?

Reply:
So, I don't see a future where we copy the target/repl from the device or launch a single device thread to check for these being equal.